mcnp/mcnp6/CMakeLists.txt

```diff
@@ -0,0 +1,193 @@
+set(MCNP_VERSION Unknown)
+if ("/${MCNP_SRC_FILES};" MATCHES "/main.F90;")
+  set(MCNP_VERSION 6.1)
+endif ()
+if ("/${MCNP_SRC_FILES};" MATCHES "/cerenkov_mod.F90;" AND
+    "/${MCNP_SRC_FILES};" MATCHES "/cgm_interface.F90;" AND
+    "/${MCNP_SRC_FILES};" MATCHES "/parse_ichoic.F90;" AND
+    "/${MCNP_SRC_FILES};" MATCHES "/reflect_light.F90;")
+  set(MCNP_VERSION 6.1.1)
+endif ()
+if (${MCNP_VERSION} STREQUAL Unknown)
+  message(FATAL_ERROR "MCNP6 source files not found in mcnp6/Source/src")
+endif (${MCNP_VERSION} STREQUAL Unknown)
+if (NOT "/${MCNP_SRC_FILES};" MATCHES "/dagmc_mod.F90;")
+  message(FATAL_ERROR "MCNP6 source version ${MCNP_VERSION} files were found, but dagmc_mod.F90 was not. Perhaps the patch was not applied?")
+endif (NOT "/${MCNP_SRC_FILES};" MATCHES "/dagmc_mod.F90;")
+message(STATUS "DAG-MCNP6 version ${MCNP_VERSION} source files found")
+
+# C compiler flags
+if (${CMAKE_C_COMPILER_ID} STREQUAL Intel)
+  #set(CMAKE_C_FLAGS "${CMAKE_Fortran_FLAGS} -mcmodel=medium")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pc64")
+else ()
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -m64")
+endif (${CMAKE_C_COMPILER_ID} STREQUAL Intel)
+
+# Fortran compiler flags
+if (${CMAKE_Fortran_COMPILER_ID} STREQUAL Intel)
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -no-vec")
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -heap-arrays 1024")
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -traceback")
+  #set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -mcmodel=medium")
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -pc64")
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -diag-disable 8290,8291,8577")
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -r8")
+else ()
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -m64")
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -mieee-fp")
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fdollar-ok")
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fno-sign-zero")
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fdefault-real-8")
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fdefault-double-8")
+endif (${CMAKE_Fortran_COMPILER_ID} STREQUAL Intel)
+
+# Linker flags
+if (${CMAKE_Fortran_COMPILER_ID} STREQUAL Intel)
+  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -static-intel")
+endif (${CMAKE_Fortran_COMPILER_ID} STREQUAL Intel)
+
+# Include directories
+include_directories(${CMAKE_CURRENT_LIST_DIR}/Source/import)
+if (${MCNP_VERSION} STREQUAL 6.1.1)
+  include_directories(${CMAKE_CURRENT_LIST_DIR}/Source/import/cgm)
+endif (${MCNP_VERSION} STREQUAL 6.1.1)
+include_directories(${PROJECT_SOURCE_DIR}/tally)
+include_directories(${HDF5_INCLUDE_DIRS})
+include_directories(${MOAB_INCLUDE_DIRS})
+
+# Preprocessor directives
+add_definitions(-DKODE="mcnp")
+if (MPI_BUILD)
+  add_definitions(-DVERS="6.mpi")
+else ()
+  add_definitions(-DVERS="6")
+endif (MPI_BUILD)
+if (${MCNP_VERSION} STREQUAL 6.1)
+  add_definitions(-DTHREAD="MCNP6")
```

Comment:
The differences are to maintain parity with the LANL versions. I copied those variables directly from `mcnp_env.F90`, which is a source file produced by the perl script `mcnp_env.pl`. The values of those variables in the CMakeLists are what appeared in `mcnp_env.F90` for versions 6.1 and 6.1.1.

Upon further investigation, the script `VC_info.gcf` contains `THREAD = MCNP6` and `REL_VERS = 1.0` for 6.1 and `THREAD = MCNP` and `REL_VERS = 6.1.1b` for 6.1.1.